bwpy/signal.py

```diff
@@ -0,0 +1,133 @@
+from . import mea_viewer
+import re
+import abc
+import functools
+import numpy as np
+import numpy.lib.stride_tricks as np_tricks
+
+
+__all__ = []
+
+
+def _transformer_factory(cls):
+    @functools.wraps(cls)
+    def transformer_factory(slice, *args, **kwargs):
+        return slice._transform(cls(*args, **kwargs))
+
+    return transformer_factory
+
+
+class Transformer(abc.ABC):
+    def __init_subclass__(cls, operator=None, **kwargs) -> None:
+        super().__init_subclass__(**kwargs)
+        name = operator or re.sub("([a-z0-9])([A-Z])", r"\1_\2", cls.__name__).lower()
+        globals()[name] = _transformer_factory(cls)
+        __all__.append(name)
+
+    def _apply_window(self, data, bin_size):
+        rows = data.shape[1]
+        cols = data.shape[2]
+        window_shape = (bin_size, rows, cols)
+        # sliding window returns more complex shape like (num_windows, 1, 1, rows, cols, bin_size)
+        # with the reshape we get rid of the unnecessary complexity (1, 1)
```

Comment:
FWIW, this is because you're using `data.shape[1]` and `data.shape[2]`, so there's always only 1 value in those dimensions per window :) Your window is multidimensional too ^_^ Nice reshape though, just wanted to remark that it isn't always unnecessary complexity.